admin/worker/upgrade_latest_version_projects.go

```diff
@@ -0,0 +1,93 @@
+package worker
+
+import (
+	"context"
+
+	"github.com/hashicorp/go-version"
+	"github.com/rilldata/rill/admin"
+	"github.com/rilldata/rill/admin/database"
+	"github.com/rilldata/rill/runtime/pkg/observability"
+	"go.uber.org/zap"
+)
+
+func (w *Worker) upgradeLatestVersionProjects(ctx context.Context) error {
+	// Resolve latest version
+	latestVersion := "latest"
+	if w.admin.VersionNumber != "" {
+		latestVersion = w.admin.VersionNumber
+	}
+
+	// Verify latest version is a valid SemVer or 'latest'
+	if latestVersion != "latest" {
+		_, err := version.NewVersion(latestVersion)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Iterate over batches of projects with 'latest' version
+	limit := 100
+	afterName := ""
+	stop := false
+	for !stop {
+		// Get batch and update iterator variables
+		projs, err := w.admin.DB.FindProjectsByVersion(ctx, "latest", afterName, limit)
+		if err != nil {
+			return err
+		}
+		if len(projs) < limit {
+			stop = true
+		}
+		if len(projs) != 0 {
+			afterName = projs[len(projs)-1].Name
+		}
+
+		// Process batch
+		for _, proj := range projs {
+			err := w.upgradeAllDeploymentsForProject(ctx, proj, latestVersion)
+			if err != nil {
+				// We log the error, but continues to the next deployment
+				w.logger.Error("upgrade latest version projects: failed to upgrade project deployments", zap.String("project", proj.ID), zap.String("version", latestVersion), observability.ZapCtx(ctx))
+			}
+		}
+	}
+
+	return nil
+}
+
+func (w *Worker) upgradeAllDeploymentsForProject(ctx context.Context, proj *database.Project, latestVersion string) error {
+	// Get all project deployments
+	depls, err := w.admin.DB.FindDeploymentsForProject(ctx, proj.ID)
+	if err != nil {
+		return err
+	}
+
+	// Get project organization, we need this to create the deployment annotations
+	org, err := w.admin.DB.FindOrganization(ctx, proj.OrganizationID)
+	if err != nil {
+		return err
+	}
+
+	for _, depl := range depls {
+		if depl.RuntimeVersion != latestVersion {
+			w.logger.Info("upgrade latest version projects: upgrading deployment", zap.String("deployment", depl.ID), zap.String("provision", depl.ProvisionID), zap.String("instance", depl.RuntimeInstanceID), zap.String("version", latestVersion), observability.ZapCtx(ctx))
```

Comment:
Same – `deployment_id`, `provision_id`, `instance_id`

Reply:
I just followed the naming convention from one of the other jobs? Should we change them to reflect the convention you mention here?
https://github.com/rilldata/rill/blob/a2614c0c4f300a2b702eeb2eaf2b83b70e0f7d50/admin/worker/reset_all_deployments.go#L48

Reply:
Updated - I also updated `reset_all_deployments` to use the same convention